scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/PostgresSpecificationRepository.java

```diff
@@ -1,25 +1,69 @@
 package gov.nasa.jpl.aerie.scheduler.server.remotes.postgres;
 
-import javax.sql.DataSource;
-
+import gov.nasa.jpl.aerie.scheduler.CardinalityGoal;
+import gov.nasa.jpl.aerie.scheduler.Goal;
 import gov.nasa.jpl.aerie.scheduler.server.exceptions.NoSuchSpecificationException;
+import gov.nasa.jpl.aerie.scheduler.server.models.GoalRecord;
+import gov.nasa.jpl.aerie.scheduler.server.models.PlanId;
+import gov.nasa.jpl.aerie.scheduler.server.models.GoalId;
 import gov.nasa.jpl.aerie.scheduler.server.models.Specification;
 import gov.nasa.jpl.aerie.scheduler.server.models.SpecificationId;
 import gov.nasa.jpl.aerie.scheduler.server.remotes.SpecificationRepository;
 import gov.nasa.jpl.aerie.scheduler.server.services.RevisionData;
 
-public record PostgresSpecificationRepository(DataSource dataSource) implements SpecificationRepository {
+import javax.sql.DataSource;
+import java.sql.SQLException;
+import java.util.stream.Collectors;
+
+public final class PostgresSpecificationRepository implements SpecificationRepository {
+  private final DataSource dataSource;
+
+  public PostgresSpecificationRepository(final DataSource dataSource) {
+    this.dataSource = dataSource;
+  }
 
   @Override
-  public Specification getSpecification(final SpecificationId specificationId) throws NoSuchSpecificationException
-  {
-    throw new UnsupportedOperationException(); // TODO stubbed method must be implemented
+  public Specification getSpecification(final SpecificationId specificationId) throws NoSuchSpecificationException {
+    try (final var connection = this.dataSource.getConnection()) {
+      try (
+          final var getSpecificationAction = new GetSpecificationAction(connection);
+          final var getSpecificationRulesAction = new GetSpecificationRulesAction(connection)
+      ) {
+        final var specificationRecord = getSpecificationAction.get(specificationId);
+        final var goals = getSpecificationRulesAction
+            .get(specificationId)
+            .stream()
+            .map(g -> new GoalRecord(
+                new GoalId(g.id()),
+                new CardinalityGoal.Builder().named(g.definition()).build()))
```

Comment:
This will not work, the builder will tell you other fields (thereExists, inPeriod...) are missing. This : 
 ```
 CardinalityGoal goal = new CardinalityGoal.Builder()
      .inPeriod(ActivityExpression.ofType(new ActivityType("")))
      .thereExistsOne(new ActivityCreationTemplate.Builder()
                          .ofType(new ActivityType(""))
                          .build())
      .named("TestCardGoal")
      .forAllTimeIn(Window.at(Duration.SECONDS))
      .owned(ChildCustody.Jointly)
      .build();
```
is the minimalistic definition of a CardinalityGoal. And it should be much more restrictive than that :)

Reply:
Ah, yes this is the part I'm the least knowledgeable on. The need I was trying to address is simply that I need to create a Goal with the name instantiated, but I don't have any other information about the goal, because it's just a placeholder until we load the goals from the JAR. Might you have a suggestion for how to better handle this?

Reply:
Either you can use the bit of code I have provided to create the minimal goal (with the appropriate name) or you will have to skip the creation of the fake goal and explicitly check the loaded jar goal name against the text content of the goal record. I have no preference as all of this is a temporary workaround.

Reply:
Alright, thanks, I'll go with the code you have provided, as I would like to keep the interfaces accurate (`Specification` contains `GoalRecord`s which contains `Goal`s), so that replacing the workaround requires minimal change. Thanks Adrien!